### ISSUE TITLE
Installer pods_project.pod_groups returns incorrect reference with :generate_multiple_pod_projects

### ISSUE THREAD

**Issue by cltnschlosser** [CocoaPods/CocoaPods#8849]
🌈
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

With `:generate_multiple_pod_projects => true`:
`installer.pods_project.pod_groups` returns reference to the sub project, not the group for the pod. The group is one level deeper. Also the return type is `Array<PBXFileReference>` vs the documented `Array<PBXGroup>`

This breaks a post install script I'm using to add a folder reference for a development pod. This folder includes configuration files that are nice to tweak from xcode, but aren't resources to be included in the final build. If there is a better way to do this, let me know.
```ruby
  installer.pods_project.pod_groups.each do |pod_group|
    if pod_group.name == "LocalPod"
      dir = pod_group.real_path.to_s + '/Operations/'
      installer.pods_project.add_file_reference(dir, pod_group, true, nil)
    end
  end
```

## What did you expect to happen?

`installer.pods_project.pod_groups` returns reference to the `PBXGroup` inside of the sub project.

## What happened instead?

`installer.pods_project.pod_groups` returns reference to the sub project with is a `PBXFileReference`.

### Stack

```
   CocoaPods : 1.7.0
        Ruby : ruby 2.3.3p222 (2016-11-21 revision 56859) [x86_64-darwin18]
    RubyGems : 3.0.3
        Host : Mac OS X 10.14.4 (18E226)
       Xcode : 10.2.1 (10E1001)
         Git : git version 2.20.1 (Apple Git-117)
Ruby lib dir : /usr/local/var/rbenv/versions/2.3.3/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ e3650707a8182edf02482a3a5dd60355b15a9db5
```

### Installation Source

```
Executable Path: /usr/local/var/rbenv/versions/2.3.3/bin/pod
```

### Plugins

```
claide-plugins        : 0.9.2
cocoapods-deintegrate : 1.0.4
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.1.0
cocoapods-trunk       : 1.3.1
cocoapods-try         : 1.1.0
slather               : 2.4.4
```

**Comment by dnkoutso** [comment]
Cc @sebastians

**Comment by sebastianv1** [comment]
Yeah `#pod_groups` might need some updating with `generate_multiple_pod_projects` since the children of the `Pods` and `Development Pods` groups are now subprojects instead of groups. 

@cltnschlosser I would recommend using the `#generated_projects` or for just the set of projects of pods, `#generated_pod_targets` on `installer` to find the group for "LocalPod" instead of just `pods_project`.

**Comment by cltnschlosser** [comment]
Thanks for the tip, I'll keep in mind if I'm able to switch to multiple xcodeprojs. We use FBSDKLoginKit, so currently blocked by the way they import internal headers from FBSDKCoreKit 🙁

**Comment by sebastianv1** [comment]
You could still use `generated_projects` today (supports both single and multi pod projects) so that when you do switch over it _should_ just work. But thanks for pointing out the issue with `#pod_groups`.

**Comment by dnkoutso** [comment]
I tried to take a stab at this. I managed to fix it but I am not happy. In multi-xcodeproj to return the pod group for subprojects means we have to call `Project.load` in order to be able to return the group reference of that project.

This can be costly. as for large projects we have to re-open _all_ of them.

This needs a different model in which all sub-projects are passed into `project.rb` after they have been created to be properly solved.

**Comment by dnkoutso** [comment]
Removing from 1.10 milestone.
